unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2968,38 +2990,123 @@ int layerMask
                 #endif
                 pathDistance += Vector3.Distance(path.corners[i], path.corners[i + 1]);
             }
-            
-
-            //disable navmesh agent
-            this.GetComponent<UnityEngine.AI.NavMeshAgent>().enabled = false;
 
             return path;
         }
 
-        public void GetShortestPathToPoint(Vector3 position, float x, float y, float z) {
-            Vector3 startPosition = position;
-            var targetPosition = new Vector3(x, y, z);
+        protected float getFloorY(float x, float start_y, float z) {
+            int layerMask = ~(1 << 10 | 1 << 9);
```

Comment:
Could you use the `LayerMask.GetMask("TargetLayer")` function to get the layers, to make it easier to read in the future?

Reply:
I did not know that `LayerMask.GetMask` existed! Done.